packages/asset/contracts/Asset.sol

```diff
@@ -37,22 +39,28 @@ contract Asset is
     mapping(address => uint16) public creatorNonces;
     // mapping of old bridged tokenId to creator nonce
     mapping(uint256 => uint16) public bridgedTokensNonces;
+    // mapping of ipfs token uri to token ids
+    mapping(string => uint256) public ipfsStringUsed;
```

Comment:
Is there a way to extract all the part related to `ipfsStringUsed` and put that in another contract that would provide a modifier `exists` / `metadataHashExists` for instance ?

Reply:
That would be the next step to extract logic into different contracts